src/app/button-page/button-page.component.ts

```diff
@@ -102,4 +125,92 @@ export class ButtonPageComponent implements OnInit, OnDestroy {
   calculateRowHeight(): string {
     return (100 / this.board.grid.rows).toString() + '%';
   }
+
+  currentScanHighlightRow(row: string[]): boolean {
+    const highlight = this.scanningModel && this.scanningModel.currentHighlight;
+    if (highlight && highlight.type === ScannableButtonRow.TYPE) {
```

Comment:
I'm fairly convinced in the back of my head that if you did a combined model in a certain way you could dispense with the type check in these three methods

Reply:
Four* methods